nengo/builder/signal.py

```diff
@@ -42,6 +42,9 @@ class Signal:
         is a view already (in which case the signal base offset will be 0
         because it starts where the view starts. That NumPy view can have
         an offset of itself).
+    allow_nan : bool, optional
```

Comment:
Would there ever be a case where someone would want to set this to True?

Reply:
Just went ahead and removed it to keep things simple, but can add it back in if you had a use case in mind!

Reply:
The only time we intentionally use NaNs that I know of is in Nengo GUI for sliders to indicate which dimensions are overridden by the GUI and which are to continue as normal. We could always do that another way though, and that situation probably still wouldn't trip the NaN error because it happens _during_ the simulation, not in the `initial_value`. So I'm fine removing it until we get a real use case.